update several statements (#2862)
pingcap/docs: sql-statements/sql-statement-set-transaction.md

* TiDB supports the ability to set a transaction as read-only in syntax only.
* The isolation levels `READ-UNCOMMITTED` and `SERIALIZABLE` are not supported.

> * Snapshot isolation (Snapshot Isolation) technology to achieve optimistic transaction `REPEATABLE-READ` isolation level, partly compatible with MySQL.

Reviewer: ```suggestion
* The `REPEATABLE-READ` isolation level is achieved through using the snapshot isolation technology, which is compatible with MySQL.
```
